rust/tool/src/esp.rs

```diff
@@ -94,31 +108,38 @@ impl EspGenerationPaths {
     }
 }
 
-fn nixos_path(path: impl AsRef<Path>, name: &str) -> Result<PathBuf> {
-    let resolved = path
-        .as_ref()
-        .read_link()
-        .unwrap_or_else(|_| path.as_ref().into());
-
-    let parent_final_component = resolved
-        .parent()
-        .and_then(|x| x.file_name())
-        .and_then(|x| x.to_str())
-        .with_context(|| format!("Failed to extract final component from: {:?}", resolved))?;
-
-    let nixos_filename = format!("{}-{}.efi", parent_final_component, name);
+fn hash<'key, T: Serialize>(inputs: impl IntoIterator<Item = (&'key str, T)>) -> Result<String> {
+    let inputs = serde_json::to_string(&BTreeMap::from_iter(inputs))
+        .context("Failed to serialize inputs")?;
+    Ok(Sha256::digest(inputs)
+        .iter()
+        .flat_map(|b| [b >> 4, b & 0xf])
```

Comment:
Please add a comment or explain what is this supposed to be doing. I imagine you want readable hash identifiers.

Reply:
(I suppose it's hexdigest conversion)